### PR TITLE
fix: fix and refactor gex app

### PR DESCRIPTION
**Pull request by Pantani**
### Description

- The Gex app was not working when we passed a custom RPC address;
- Add gex integration tests;
- Use `ignite/gex@v1.0.0` instead `cosmos/gex`;

### CODE REVIEW

File: explorer/pkg/xurl/xurl_test.go [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+			addr: "http://localhost:4000",
+			want: &url.URL{Host: "localhost:4000", Scheme: schemeHTTP},
+		},
+		{
+			name: "custom ssl",
+			addr: "https://localhost:4005",
+			want: &url.URL{Host: "localhost:4005", Scheme: schemeHTTPS},
+		},
+		{
+			name: "no schema and port",
```
Comment (jeronimoalbi) [inline explorer/pkg/xurl/xurl_test.go:39]:
```suggestion
			name: "no scheme and port",
```

File: explorer/pkg/xurl/xurl_test.go [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+			addr: "localhost",
+			want: &url.URL{Host: "localhost:80", Scheme: schemeHTTP},
+		},
+		{
+			name: "no schema",
```
Comment (jeronimoalbi) [inline explorer/pkg/xurl/xurl_test.go:44]:
```suggestion
			name: "no scheme",
```

File: explorer/pkg/xurl/xurl_test.go [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+				require.Equal(t, tt.err.Error(), err.Error())
+				return
+			}
+			require.NoError(t, err)
+			require.EqualValues(t, tt.want, addr)
+		})
+	}
+}
+
+func Test_addressPort(t *testing.T) {
```
Comment (jeronimoalbi) [inline explorer/pkg/xurl/xurl_test.go:69]: Maybe we could omit the private function tests by making sure that these cases are included in the public function tests

File: explorer/CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,11 @@
 # Explorer App Changelog
 
+## [`v0.3.0`](https://github.com/ignite/apps/releases/tag/explorer/v0.3.0)
+
+* The Gex app was not working when we passed a custom RPC address.
```
Comment (salmad3) [inline explorer/CHANGELOG.md:5]:
```suggestion
* The Gex app was malfunctioning when passing a custom RPC address.
```

**Review by salmad3 — DISMISSED** [review]
Concept ACK

File: explorer/go.mod [hunk 5]
```diff
@@ -1,11 +1,11 @@
 module github.com/ignite/apps/explorer
 
-go 1.21.1
+go 1.22.2
```
Comment (jeronimoalbi) [inline explorer/go.mod:3]: If possible it might be good to keep the version low, unless a dependency or new feature requires to bump it

**Review by jeronimoalbi — APPROVED** [review]

**Review by jeronimoalbi — APPROVED** [review]

**Pull request merged**